test/Preact10ShallowDiff/Preact10ShallowDiffMemo_test.tsx

```diff
@@ -0,0 +1,1606 @@
+  it.skip('should filter context by contextTypes', () => {
+    const SimpleComponent = memo(
+      class SimpleComponent extends Component {
+        static contextTypes: any = {
+          foo: PropTypes.string,
+        };
+        render() {
+          return <div>{`${this.context.foo}:${this.context.bar}`}</div>;
+        }
+      }
+    );
+
+    const shallowRenderer = createRenderer();
+    const result = shallowRenderer.render(<SimpleComponent />, {
+      foo: 'foo',
+      bar: 'bar',
+    });
+    expect(result).toEqual(<div>foo:undefined</div>);
+  });
+
+  // Preact doesn't support this contextTypes
+  it.skip('can fail context when shallowly rendering', () => {
+    const SimpleComponent = memo(
+      class SimpleComponent extends Component {
+        static contextTypes: any = {
+          name: PropTypes.string.isRequired,
+        };
+
+        render() {
+          return <div>{this.context.name}</div>;
+        }
+      }
+    );
+
+    const shallowRenderer = createRenderer();
+    expect(() => shallowRenderer.render(<SimpleComponent />)).toErrorDev(
+      'Warning: Failed context type: The context `name` is marked as ' +
+        'required in `SimpleComponent`, but its value is `undefined`.\n' +
+        '    in SimpleComponent (at **)'
+    );
+  });
+
+  // Preact only validates prop types when preact/debug is included
+  it.skip('should warn about propTypes (but only once)', () => {
+    const SimpleComponent = memo(
+      class SimpleComponent extends Component<{ name: any }> {
+        static propTypes: any = {
+          name: PropTypes.string.isRequired,
+        };
+
+        render() {
+          return createElement('div', null, this.props.name);
+        }
+      }
+    );
+
+    const shallowRenderer = createRenderer();
+    expect(() =>
+      shallowRenderer.render(createElement(SimpleComponent, { name: 123 }))
+    ).toErrorDev(
+      'Warning: Failed prop type: Invalid prop `name` of type `number` ' +
+        'supplied to `SimpleComponent`, expected `string`.\n' +
+        '    in SimpleComponent'
+    );
+  });
+
+  it('should enable rendering of cloned element', () => {
+    type Props = { foo: string };
+    type State = { bar: string };
+    const SimpleComponent = memo(
+      class SimpleComponent extends Component<Props, State> {
+        constructor(props: Props) {
+          super(props);
+
+          this.state = {
+            bar: 'bar',
+          };
+        }
+
+        render() {
+          return <div>{`${this.props.foo}:${this.state.bar}`}</div>;
+        }
+      }
+    );
+
+    const shallowRenderer = createRenderer();
+    const el = <SimpleComponent foo="foo" />;
+    let result = shallowRenderer.render(el);
+    expect(result).toEqual(<div>foo:bar</div>);
+
+    const cloned = cloneElement(el, { foo: 'baz' });
+    result = shallowRenderer.render(cloned);
+    expect(result).toEqual(<div>baz:bar</div>);
+  });
+
+  // TODO: Renable when preactjs/preact#3806 is released
```

Comment:
I see this PR has been merged, so I assume this will get updated soon?

Reply:
Yup. I'll update it whenever the next version is published (maybe in a week or two?)